frontend/src/pages/CollaborationReportForm/index.js

```diff
@@ -302,25 +357,129 @@ function CollaborationReport({ match, location, region }) {
+        const savedReport = await createReport({
+          ...data,
+          regionId: formData.regionId,
+          approverUserIds: approverIds,
+          version: 2,
+        });
+
+        if (!savedReport) {
+          throw new Error('Report not found');
+        }
+
+        reportId.current = savedReport.id;
+
+        cleanupLocalStorage('new', savedReport.id);
+
+        window.history.replaceState(null, null, `/collaboration-reports/${savedReport.id}/${currentPage}`);
+
+        setConnectionActive(true);
+        updateCreatorRoleWithName(savedReport.creatorNameWithRole);
+      } else {
+        const updatedReport = await formatReportWithSaveBeforeConversion(
+          data,
+          formData,
+          user,
+          userHasOneRole,
+          reportId,
+          approverIds,
+          forceUpdate,
+        );
+
+        updateFormData(updatedReport, true);
+        setConnectionActive(true);
+        updateCreatorRoleWithName(updatedReport.creatorNameWithRole);
+      }
+    } catch (e) {
+      setConnectionActiveWithError(error, setConnectionActive);
+    }
+  };
+
+  const setSavingLoadScreen = (isAutoSave = false) => {
+    if (!isAutoSave && !isAppLoading) {
+      setAppLoadingText('Saving');
+      setIsAppLoading(true);
+    }
+  };
+
+  const onSaveDraft = async (isAutoSave = false, forceUpdate = false) => {
+    try {
+      // Turn on loading screen
+      setSavingLoadScreen(isAutoSave);
+
+      // If not editable, don't show the loading screen
+      if (!editable) {
+        setIsAppLoading(false);
+        return;
+      }
+
+      // Get the current form data
+      const { status, ...formValues } = hookForm.getValues();
+      // TODO: Add 'pageState' and newNavigatorState to the saved data
+      const data = { ...formData, ...formValues };
+
+      // Clear the previous error message if there is one
+      updateErrorMessage();
+
+      // save the form data to the server
+      await onSave(data, forceUpdate);
+
+      // Update the last saved time
+      updateLastSaveTime(moment());
 
-    // const page = pages.find((p) => p.position === position);
-    // const newPath = `/collaboration-reports/${reportId.current}/${page.path}`;
-    // history.push(newPath, state);
+      // show the saved draft message
+      updateShowSavedDraft(true);
+    } catch (e) {
+      updateErrorMessage('A network error has prevented us from saving your collaboration report to our database. You work is safely saved to your web browser in the meantime.');
+    } finally {
+      setIsAppLoading(false);
+    }
   };
 
-  // ===========================================================================//
-  //
-  // NOTE: onSave, onFormSubmit, onReview, and onResetToDraft go here eventually
-  //
-  // ===========================================================================//
+  const onFormSubmit = async (data) => {
+    const approverIds = data.approvers.map((a) => a.user.id);
```

Comment:
Should we check if `data.approvers` exist here before calling `map`?
e.g. a check similar to the one we have in `onSave`